lagent/llms/lmdeploy_wrapper.py

```diff
@@ -226,11 +233,32 @@ def __init__(self,
                  tp: int = 1,
                  pipeline_cfg=dict(),
                  **kwargs):
-
+        import lmdeploy
+        from lmdeploy import ChatTemplateConfig, TurbomindEngineConfig, pipeline, version_info
+
+        self.str_version = lmdeploy.__version__
+        self.version = version_info
+        self.do_sample = kwargs.pop('do_sample', None)
+        if self.do_sample is not None and self.version < (0, 6, 0):
+            raise RuntimeError(
+                '`do_sample` parameter is not supported by lmdeploy until '
+                f'v0.6.0, but currently using lmdeloy {self.str_version}')
         super().__init__(path=path, **kwargs)
-        from lmdeploy import pipeline
+        backend_config = copy.deepcopy(pipeline_cfg)
+        backend_config.update(tp=tp)
+        backend_config = {
+            k: v
+            for k, v in backend_config.items()
+            if hasattr(TurbomindEngineConfig, k)
+        }
+        backend_config = TurbomindEngineConfig(**backend_config)
```

Comment:
How to handle PytorchEngineConfig?

Reply:
pipeline will fallback to PyTorch Engine if the model is not supported by TurboMind engine